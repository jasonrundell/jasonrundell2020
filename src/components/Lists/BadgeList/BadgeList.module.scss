@import '../../../../src/styles/main.scss';

.root {
  display: flex;
  flex-direction: column;
  list-style: none;
  padding: 0;

  .list-item {
    display: block;
    margin-bottom: $sizeSmall;
  }
}

.horizontal {
  flex-direction: row;
  flex-wrap: wrap;

  .list-item {
    display: flex;
    align-items: center;
    margin-right: $sizeSmall;
  }
}
