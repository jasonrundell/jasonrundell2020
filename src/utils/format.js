/**
 *  Letter case conversion functions
 *  [Letter case](https://en.wikipedia.org/wiki/Letter_case#Special_case_styles)
 *
 *  Converts only letter case-separated or hyphen-delimited strings
 */

/**
 *  Converts hyphen-delimited `string` to camel case
 *
 *  @example
 *  'foo-bar' => 'fooBar'
 *  'foo--bar' => 'fooBar'
 *
 *  @param {string} string
 */
function toCamelCase(string) {
  return string.replace(/(-+)([a-z])/g, ($0, $1, $2) => $2.toUpperCase())
}

/**
 *  Converts hyphen-delimited `string` to pascal case (aka upper camel case)
 *
 *  @example
 *  'foo-bar' => 'FooBar'
 *  'foo--bar' => 'FooBar'
 *
 *  @param {string} string
 */
function toPascalCase(string) {
  return toCamelCase(string.replace(/^[a-z]/, $0 => $0.toUpperCase()))
}

export { toCamelCase, toPascalCase }
