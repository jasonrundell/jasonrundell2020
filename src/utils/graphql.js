export function edgesToObject(edgesData) {
  const nodes = edgesData.map(item => {
    return item.node
  })
  const finalObject = nodes.map(item => {
    return item
  })

  return finalObject
}
