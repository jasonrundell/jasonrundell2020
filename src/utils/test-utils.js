export function catchWarnings() {
  const originalConsoleError = global.console.error
  beforeEach(() => {
    /**
     * I Want to know if values passed to props fail the type check
     * */
    global.console.error = (...args) => {
      const propTypeFailures = [/Warning:/] // e.g. [/Invalid prop/, /Failed prop type/]

      if (propTypeFailures.some(p => p.test(args[0]))) {
        throw new Error(args[0])
      }

      originalConsoleError(...args)
    }
  })
}

/**
 * Mix an match help from:
 * https://daveceddia.com/testing-for-proptypes-errors/
 * https://www.jackfranklin.co.uk/blog/failing-tests-on-react-proptypes/
 */
