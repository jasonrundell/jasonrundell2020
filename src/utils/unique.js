const onlyUnique = (value, index, self) => {
  return self.indexOf(value) === index
}

const generatePassword = ({
  passwordLength = 16,
  charset = 'abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789!@#$%^&*()_-=~`',
}) => {
  let retVal = ''
  for (var i = 0, n = charset.length; i < passwordLength; ++i) {
    retVal += charset.charAt(Math.floor(Math.random() * n))
  }
  return retVal
}

export { onlyUnique, generatePassword }
