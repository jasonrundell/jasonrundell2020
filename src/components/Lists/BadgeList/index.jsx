import React from 'react'
import PropTypes from 'prop-types'

import Badge from '../../Badge'
import Layout from '../../Layout'

import { classNames } from '../../../utils/css'

import styles from './BadgeList.module.scss'

const { Box } = Layout

const BadgeList = ({ list, isHorizontal, isInverse }) => {
  const className = classNames(styles.root, isHorizontal && styles.horizontal)
  return (
    <ul className={className}>
      {list.map(item => (
        <li className={styles.listItem} key={item.id}>
          <Box isTight>
            <Badge icon={item.icon} isInverse={isInverse} />
          </Box>
          <Box isTight>
            <span>{item.label}</span>
          </Box>
        </li>
      ))}
    </ul>
  )
}

BadgeList.defaultProps = {
  isHorizontal: false,
  isInverse: false,
}

BadgeList.propTypes = {
  list: PropTypes.array.isRequired,
  isHorizontal: PropTypes.bool,
  isInverse: PropTypes.bool,
}

export default BadgeList
