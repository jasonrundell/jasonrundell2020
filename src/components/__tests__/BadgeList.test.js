import React from 'react'
import { render, unmountComponentAtNode } from 'react-dom'
import { act } from 'react-dom/test-utils'
import renderer from 'react-test-renderer'
import { catchWarnings } from '../../utils/test-utils'

import BadgeList from '../Lists/BadgeList'

catchWarnings()

let container = null
beforeEach(() => {
  container = document.createElement('div')
  document.body.appendChild(container)
})

afterEach(() => {
  unmountComponentAtNode(container)
  container.remove()
  container = null
})

const listData = [
  {
    id: 'eb882bd6-c391-5613-9795-2dd6cd134e54',
    icon: '⭐',
    label: 'Star',
  },
  {
    id: 'f4f7b181-cc7d-57ce-9d59-15a9916fbec1',
    icon: '😀',
    label: 'Smiley',
  },
  {
    id: 'd4b52d23-aa82-5539-badb-48382003d372',
    icon: '💫',
    label: 'Dizzy',
  },
]

describe('BadgeList component', () => {
  it('BadgeList renders correctly', () => {
    const tree = renderer
      .create(
        <div>
          <BadgeList list={listData} />
        </div>
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
