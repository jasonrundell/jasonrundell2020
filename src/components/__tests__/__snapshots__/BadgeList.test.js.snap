// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BadgeList component BadgeList renders correctly 1`] = `
<div>
  <ul
    className="root"
  >
    <li
      className="listItem"
    >
      <div
        className="root tight"
      >
        <div
          className="root"
        >
          <div
            className="root tight"
          >
            ⭐
          </div>
        </div>
      </div>
      <div
        className="root tight"
      >
        <span>
          Star
        </span>
      </div>
    </li>
    <li
      className="listItem"
    >
      <div
        className="root tight"
      >
        <div
          className="root"
        >
          <div
            className="root tight"
          >
            😀
          </div>
        </div>
      </div>
      <div
        className="root tight"
      >
        <span>
          Smiley
        </span>
      </div>
    </li>
    <li
      className="listItem"
    >
      <div
        className="root tight"
      >
        <div
          className="root"
        >
          <div
            className="root tight"
          >
            💫
          </div>
        </div>
      </div>
      <div
        className="root tight"
      >
        <span>
          Dizzy
        </span>
      </div>
    </li>
  </ul>
</div>
`;
